### DISCUSSION TITLE
Interpreting Window Effects

### DISCUSSION THREAD

**Discussion by non87** [stocnet/goldfish#45] [Q&A] [answered]
Hello everyone, I am a new user of Goldfish. As a first thing, thank you very much for this wonderful package!

I am trying to use DyNAM model for my research, where I will have to estimate the effect of certain kind of dyadic events on the probability of other kinds of dyadic events. In particular, I have data about artists' group exhibitions and art auctions. I want to test if exhibiting together translate into selling together. 

My best attempt at specifying such a model is the following (I am using the _Social evolution_ data for reproducibility):

```library(goldfish)
data("Social_Evolution")
set.seed(12345)

actors <- defineNodes(nodes = actors)
callNetwork <- defineNetwork(nodes = actors, directed = TRUE)
callNetwork <- linkEvents(x = callNetwork, changeEvent = calls,
                          nodes = actors)
callsDependent <- defineDependentEvents(events = calls, nodes = actors,
                                        defaultNetwork = callNetwork)
# Create a second network of calls to be used in the model specification
calls2 <- calls
# create a time differential between the two call networks
calls2$time <- calls2$time - 1500
# Subsample and re-order
calls2 <- calls2[sample.int(nrow(calls2), 100, replace = F), ]
calls2 <- calls2[order(calls2$time),]
callNetwork2 <- defineNetwork(nodes = actors, directed = TRUE)
callNetwork2 <- linkEvents(x = callNetwork2, changeEvent = calls2,
                          nodes = actors)

# Notice the last term of the formula
modNet <- estimate(callsDependent ~ inertia + recip(callNetwork) + trans + same(actors$floor) 
                  + same(actors$gradeType) + alter(actors$gradeType) + tie(callNetwork2, window=1500),
                  model = "DyNAM", subModel = "choice")

summary(modNet)
```

The key part of the code is the effect specification `tie(callNetwork2, window=1500)`. I interpret it as the effect on the choice due to the event happening on the covariate network `callNetwork2` in the 1500 seconds before the choice being modeled. This is exactly what I want, but I would *really* appreciate if a more expert user could vet the code (and perhaps correct it). Thanks again.

**Comment by auzaheta** [comment, marked as answer]
Hi @non87, thanks for posting your question and the good vibes about the package.

Short answer: Your interpretation is correct. 
Long answer: [Figure 2 on Stadfeld C. and Per B.](https://doi.org/10.15195/v4.a14) has a nice representation of how the window effects look like in the timeline of events. `goldfish` does a quite straightforward representation of this creating a new events data frame that adds the dissolution (deletion) of the event happening on a plus window size time after the event. 
You can look at the Global environment to a new object called something like `calls2_1500`. You will notice that every event on calls2 would have a corresponding event with an increment $-1$ at the time $time + 1500$. If a dependent event happens between the 1500 seconds after an event in `calls2`,  then the corresponding change statistics would include the event in the covariate network as part of the explanatory effects during estimation.

```r
> head(calls2_1500, 2)
        time   sender receiver increment
1 1220733470 Actor 72 Actor 50         1
2 1220734970 Actor 72 Actor 50        -1
```

As a remark of use of the window parameter: if the event time in your data is in seconds or an `R` date format, you can use shortcuts to define the windows using strings stating the preferred unit of time. In your example, `window = "25 minutes"`  would give the same result.

It looks like it is worthed to improve the documentation regarding window effects and the effect interpretation. Thanks for pointing that out. Maybe a change in the title of this question might help other users to find this info. Something on the lines of "interpreting window effects" might do the trick.

I hope the answer is clear. Let me know if you would like to clarify something else.
